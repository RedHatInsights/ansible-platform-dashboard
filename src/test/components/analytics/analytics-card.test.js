import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { MemoryRouter } from 'react-router-dom';
import notificationsMiddleware from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
import analyticsReducer, { analyticsInitialState } from '../../../redux/reducers/analytics-reducer';
import AnalyticsCard from '../../../components/analytics/analytics-card';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { applyReducerHash, ReducerRegistry } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';
import '../../../helpers/shared/__mocks__/user-login';

const ComponentWrapper = ({ store, initialEntries = [ '/ansible-dashboard' ], children }) => (
  <IntlProvider locale="en">
    <Provider store={ store } >
      <MemoryRouter initialEntries={ initialEntries }>
        { children }
      </MemoryRouter>
    </Provider>
  </IntlProvider>
);

describe('<AnalyticsCard />', () => {
  let initialProps;
  let initialState;
  const middlewares = [thunk, promiseMiddleware, notificationsMiddleware()];
  let mockStore;

  beforeEach(() => {
    global.insights = {
      chrome: {
        identifyApp: jest.fn(),
        init: jest.fn(),
        auth: {
          getUser: () => Promise.resolve(true)
        }
      }
    };
    initialProps = {};
    mockStore = configureStore(middlewares);
    initialState = {
      i18nReducer: {
        formatMessage: ({ defaultMessage }) => defaultMessage
      },
      analyticsReducer: { ...analyticsInitialState, isLoading: false }
    };
  });

  it('should render correctly', async ()=> {
    const store = mockStore(initialState);
    const registry = new ReducerRegistry({}, [ thunk, promiseMiddleware ]);
    registry.register({ analyticsReducer: applyReducerHash(analyticsReducer, analyticsInitialState) });

    const { asFragment } = render(<ComponentWrapper store={ store }><AnalyticsCard { ...initialProps } /></ComponentWrapper>);

    await waitFor(() => expect(() => screen.getByLabelText('Contents')).toThrow());

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly in loading state', async () => {
    const store = mockStore(initialState);
    const { asFragment } = render(<ComponentWrapper store={ store }><AnalyticsCard { ...initialProps } isLoading={ true } /></ComponentWrapper>);
    expect(asFragment()).toMatchSnapshot();

    await waitFor(() => expect(() => screen.getByLabelText('Contents')).toThrow());
  });
});
