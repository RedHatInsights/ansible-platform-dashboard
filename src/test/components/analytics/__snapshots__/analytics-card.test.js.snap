// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnalyticsCard /> should render correctly 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card ans-c-card-dashboard"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <h3
        class="pf-c-title pf-m-lg"
        data-ouia-component-id="OUIA-Generated-Title-1"
        data-ouia-component-type="PF4/Title"
        data-ouia-safe="true"
      >
        Insights Ansible Automation Platform
      </h3>
    </div>
    <div
      class="pf-c-card__body pf-u-pb-sm"
    >
      <div
        class="pf-l-flex pf-m-column"
      >
        <div
          class=""
        >
          <div
            class="pf-l-flex ans-l-flex ans-l-flex-automation-analytics-card"
          >
            <div
              class=""
            >
              <div
                class="pf-l-flex  ans-l-flex ans-l-flex-automation-analytics-info"
              >
                <div
                  class=""
                >
                  <p
                    class=""
                    data-ouia-component-id="OUIA-Generated-Text-1"
                    data-ouia-component-type="PF4/Text"
                    data-ouia-safe="true"
                    data-pf-content="true"
                  >
                    Find and use content that is supported by Red Hat and our partners to deliver reassurance for the most demanding environments
                  </p>
                  <br />
                </div>
                <div
                  class=""
                >
                  <h4
                    class="pf-c-title pf-m-md"
                    data-ouia-component-id="OUIA-Generated-Title-2"
                    data-ouia-component-type="PF4/Title"
                    data-ouia-safe="true"
                  >
                    Clusters
                  </h4>
                  <br />
                  <dl
                    class="pf-c-description-list pf-m-1-col pf-c-description-list.pf-m-vertical"
                  >
                    <div
                      class="pf-c-description-list__group"
                    >
                      <dt
                        class="pf-c-description-list__term"
                      >
                        <span
                          class="pf-c-description-list__text"
                        >
                          0
                        </span>
                      </dt>
                      <dd
                        class="pf-c-description-list__description"
                      >
                        <div
                          class="pf-c-description-list__text"
                        >
                          <button
                            component="a"
                            href="/ansible/insights/clusters"
                          >
                            Total clusters
                          </button>
                          <p>
                            variant: link
                          </p>
                          <p
                            id="labelling-id"
                          >
                            Test label
                          </p>
                          <p>
                            isInline: undefined
                          </p>
                          <p>
                            iconPosition: undefined
                          </p>
                          <div
                            data-testid="icon"
                          />
                        </div>
                      </dd>
                    </div>
                  </dl>
                </div>
              </div>
            </div>
            <hr
              class="pf-c-divider"
            />
            <div
              class=""
            >
              <h4
                class="pf-c-title pf-m-md"
                data-ouia-component-id="OUIA-Generated-Title-3"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                Notifications
              </h4>
              <br />
              <dl
                class="pf-c-description-list pf-m-2-col pf-c-description-list.pf-m-vertical"
              >
                <div
                  class="pf-c-description-list__group"
                >
                  <dt
                    class="pf-c-description-list__term"
                  >
                    <span
                      class="pf-c-description-list__text"
                    >
                      0
                    </span>
                  </dt>
                  <dd
                    class="pf-c-description-list__description"
                  >
                    <div
                      class="pf-c-description-list__text"
                    >
                      <span
                        class="pf-c-label pf-m-red"
                      >
                        <a
                          class="pf-c-label__content"
                          href="/ansible/automation-analytics/notifications"
                        >
                          <span
                            class="pf-c-label__icon"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                              />
                            </svg>
                          </span>
                          <span
                            class="pf-c-label__text"
                          >
                            Critical
                          </span>
                        </a>
                      </span>
                    </div>
                  </dd>
                </div>
                <div
                  class="pf-c-description-list__group"
                >
                  <dt
                    class="pf-c-description-list__term"
                  >
                    <span
                      class="pf-c-description-list__text"
                    >
                      0
                    </span>
                  </dt>
                  <dd
                    class="pf-c-description-list__description"
                  >
                    <div
                      class="pf-c-description-list__text"
                    >
                      <span
                        class="pf-c-label pf-m-orange"
                      >
                        <a
                          class="pf-c-label__content"
                          href="/ansible/automation-analytics/notifications"
                        >
                          <span
                            class="pf-c-label__icon"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 1088 1024"
                              width="1em"
                            >
                              <path
                                d="M1057.10141,663.5 L845.101405,215.4 C787.101405,71.8 665.401405,0 542.901405,0 C420.201405,0 296.701405,71.9 235.001405,215.6 L31.7014051,648.5 C10.4014051,700 -0.0985948775,752.3 0.000697596367,800.8 C0.301405123,924.8 70.2014051,1024 209.101405,1024 L868.401405,1024 C1005.80141,1024 1087.70141,918.6 1088.00215,795.5 C1088.10141,752.4 1078.20141,707.2 1057.10141,663.5 Z M959.401405,800.3 C958.701405,822.9 952.901405,843.5 942.601405,859.7 C926.801405,884.6 902.601405,896.7 868.301405,896.7 L209.101405,896.7 C191.201405,896.7 176.601405,893.8 165.401405,888.2 C157.301405,884 150.801405,878.4 145.401405,870.3 C135.101405,855 129.101405,832 128.401405,805.6 C127.601405,772.8 134.901405,736.5 149.401405,700.5 L353.001405,266.7 C363.201405,242.9 376.101405,221.5 391.101405,203.2 C404.801405,186.6 420.301405,172.4 437.401405,161.1 C469.201405,139.9 505.701405,128.8 542.901405,128.8 C579.701405,128.8 615.401405,139.8 646.001405,160.5 C662.401405,171.6 677.101405,185.4 690.101405,201.6 C704.501405,219.6 716.401405,240.6 725.901405,264 L940.901405,718.9 L941.101405,719.3 L941.301405,719.7 C953.901405,746 960.201405,773.9 959.401405,800.3 Z M586.601405,832 L501.301405,832 C489.501405,831.8 480.201405,821.5 480.001405,808.7 L480.001405,727.3 C480.201405,714.5 489.601405,704.3 501.301405,704 L586.601405,704 C598.401405,704.2 607.701405,714.5 607.901405,727.3 L607.901405,808.7 L608.001405,808.7 C607.701405,821.5 598.301405,831.8 586.601405,832 M639.901405,290.7 L613.201405,610.4 C611.801405,626.9 598.001405,640 581.301405,640 L506.601405,640 C490.001405,640 476.101405,627.2 474.701405,610.7 L448.101405,291 C446.501405,272.3 461.301405,256.3 480.001405,256.3 L608.001405,256 C626.701405,256 641.401405,272 639.901405,290.7"
                              />
                            </svg>
                          </span>
                          <span
                            class="pf-c-label__text"
                          >
                            Warnings
                          </span>
                        </a>
                      </span>
                    </div>
                  </dd>
                </div>
              </dl>
            </div>
          </div>
        </div>
        <div
          class=""
        >
          <div
            class=""
          >
            <h4
              class="pf-c-title pf-m-md"
              data-ouia-component-id="OUIA-Generated-Title-4"
              data-ouia-component-type="PF4/Title"
              data-ouia-safe="true"
            >
              Job status
            </h4>
            <br />
            <div
              class="pf-l-flex pf-m-column"
            >
              <div
                class="pf-u-m-0"
              >
                <div
                  class="pf-c-chart"
                  style="height: 100%; width: 100%; user-select: none; pointer-events: none; position: relative;"
                >
                  <svg
                    aria-describedby="victory-container-1-desc"
                    aria-labelledby="victory-container-1-title"
                    height="225"
                    role="img"
                    style="pointer-events: all; width: 100%; height: 100%;"
                    viewBox="0 0 1024 225"
                    width="1024"
                  >
                    <title
                      id="victory-container-1-title"
                    >
                      Jobs across clusters
                    </title>
                    <desc
                      id="victory-container-1-desc"
                    >
                      Jobs across clusters
                    </desc>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="60"
                        x2="1004"
                        y1="165"
                        y2="165"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        id="chart-axis-0-axisLabel-0"
                        x="532"
                        y="206.97"
                      >
                        <tspan
                          dx="0"
                          dy="0"
                          style="padding: 30px; font-size: 14px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; stroke: transparent; text-anchor: middle; fill: #252525;"
                          text-anchor="middle"
                          x="532"
                        >
                          Time: Day
                        </tspan>
                      </text>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="242.75314283074016"
                          x2="242.75314283074016"
                          y1="165"
                          y2="170"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-0-tickLabels-0"
                          x="242.75314283074016"
                          y="188.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="middle"
                            x="242.75314283074016"
                          >
                            0.2
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="425.5062856614803"
                          x2="425.5062856614803"
                          y1="165"
                          y2="170"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-0-tickLabels-1"
                          x="425.5062856614803"
                          y="188.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="middle"
                            x="425.5062856614803"
                          >
                            0.4
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="608.2594284922204"
                          x2="608.2594284922204"
                          y1="165"
                          y2="170"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-0-tickLabels-2"
                          x="608.2594284922204"
                          y="188.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="middle"
                            x="608.2594284922204"
                          >
                            0.6
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="791.0125713229606"
                          x2="791.0125713229606"
                          y1="165"
                          y2="170"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-0-tickLabels-3"
                          x="791.0125713229606"
                          y="188.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="middle"
                            x="791.0125713229606"
                          >
                            0.8
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="973.7657141537007"
                          x2="973.7657141537007"
                          y1="165"
                          y2="170"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-0-tickLabels-4"
                          x="973.7657141537007"
                          y="188.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="middle"
                            x="973.7657141537007"
                          >
                            1
                          </tspan>
                        </text>
                      </g>
                    </g>
                    <g
                      role="presentation"
                    >
                      <line
                        role="presentation"
                        shape-rendering="auto"
                        style="stroke: #d2d2d2; fill: transparent; stroke-width: 1; stroke-linecap: round; stroke-linejoin: round;"
                        vector-effect="non-scaling-stroke"
                        x1="60"
                        x2="60"
                        y1="20"
                        y2="165"
                      />
                      <text
                        direction="inherit"
                        dx="0"
                        id="chart-axis-1-axisLabel-0"
                        transform="rotate(-90,15,92.5)"
                        x="15"
                        y="90.47"
                      >
                        <tspan
                          dx="0"
                          dy="0"
                          style="padding: 45px; font-size: 14px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; stroke: transparent; text-anchor: middle; fill: #252525;"
                          text-anchor="middle"
                          x="15"
                        >
                          Jobs across clusters
                        </tspan>
                      </text>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: none; pointer-events: painted; stroke-linecap: round; stroke-linejoin: round;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="1004"
                          y1="136"
                          y2="136"
                        />
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="55"
                          y1="136"
                          y2="136"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-1-tickLabels-0"
                          x="45"
                          y="139.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="end"
                            x="45"
                          >
                            0
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: none; pointer-events: painted; stroke-linecap: round; stroke-linejoin: round;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="1004"
                          y1="107"
                          y2="107"
                        />
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="55"
                          y1="107"
                          y2="107"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-1-tickLabels-1"
                          x="45"
                          y="110.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="end"
                            x="45"
                          >
                            0
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: none; pointer-events: painted; stroke-linecap: round; stroke-linejoin: round;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="1004"
                          y1="78"
                          y2="78"
                        />
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="55"
                          y1="78"
                          y2="78"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-1-tickLabels-2"
                          x="45"
                          y="81.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="end"
                            x="45"
                          >
                            1
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: none; pointer-events: painted; stroke-linecap: round; stroke-linejoin: round;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="1004"
                          y1="48.99999999999999"
                          y2="48.99999999999999"
                        />
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="55"
                          y1="48.99999999999999"
                          y2="48.99999999999999"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-1-tickLabels-3"
                          x="45"
                          y="52.54999999999999"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="end"
                            x="45"
                          >
                            1
                          </tspan>
                        </text>
                      </g>
                      <g
                        role="presentation"
                      >
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: none; pointer-events: painted; stroke-linecap: round; stroke-linejoin: round;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="1004"
                          y1="20"
                          y2="20"
                        />
                        <line
                          role="presentation"
                          shape-rendering="auto"
                          style="stroke: #d2d2d2; fill: transparent; size: 5px; stroke-linecap: round; stroke-linejoin: round; stroke-width: 1;"
                          vector-effect="non-scaling-stroke"
                          x1="60"
                          x2="55"
                          y1="20"
                          y2="20"
                        />
                        <text
                          direction="inherit"
                          dx="0"
                          id="chart-axis-1-tickLabels-4"
                          x="45"
                          y="23.55"
                        >
                          <tspan
                            dx="0"
                            dy="0"
                            style="font-size: 10px; font-family: \\"RedHatText\\",\\"Overpass\\",overpass,helvetica,arial,sans-serif; letter-spacing: normal; padding: 10px; stroke: transparent; fill: #4f5255;"
                            text-anchor="end"
                            x="45"
                          >
                            1
                          </tspan>
                        </text>
                      </g>
                    </g>
                    <g>
                      <g
                        role="presentation"
                      >
                        <path
                          d="M 966.8297785154242, 165 
A 0 0 0 0 1, 966.8297785154242, 165 
L 966.8297785154242, 20 
A 0 0 0 0 1, 966.8297785154242, 20 
L 976.8297785154242, 20 
A 0 0 0 0 1, 976.8297785154242, 20 
L 976.8297785154242, 165 
A 0 0 0 0 1, 976.8297785154242, 165 
 z"
                          index="0"
                          role="presentation"
                          shape-rendering="auto"
                          style="fill: #38812f; stroke: none; stroke-width: 1; padding: 8px;"
                        />
                      </g>
                      <g
                        role="presentation"
                      >
                        <path
                          d="M 966.8297785154242, 165 
A 0 0 0 0 1, 966.8297785154242, 165 
L 966.8297785154242, 20 
A 0 0 0 0 1, 966.8297785154242, 20 
L 976.8297785154242, 20 
A 0 0 0 0 1, 976.8297785154242, 20 
L 976.8297785154242, 165 
A 0 0 0 0 1, 976.8297785154242, 165 
 z"
                          index="0"
                          role="presentation"
                          shape-rendering="auto"
                          style="fill: #7d1007; stroke: none; stroke-width: 1; padding: 8px;"
                        />
                      </g>
                    </g>
                  </svg>
                  <div
                    style="z-index: 99; position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
                  >
                    <svg
                      height="225"
                      style="overflow: visible; width: 100%; height: 100%;"
                      viewBox="0 0 1024 225"
                      width="1024"
                    />
                  </div>
                </div>
              </div>
              <div
                class="pf-u-m-0 pf-u-pt-0 pf-u-pb-0 pf-u-pl-lg"
              >
                <div
                  class="pf-l-bullseye"
                >
                  <button
                    component="a"
                    href="/ansible/insights/job-explorer"
                  >
                    View Job Explorer
                  </button>
                  <p>
                    variant: link
                  </p>
                  <p
                    id="labelling-id"
                  >
                    Test label
                  </p>
                  <p>
                    isInline: undefined
                  </p>
                  <p>
                    iconPosition: undefined
                  </p>
                  <div
                    data-testid="icon"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </article>
</DocumentFragment>
`;

exports[`<AnalyticsCard /> should render correctly in loading state 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card ans-c-card-dashboard"
    data-ouia-component-id="OUIA-Generated-Card-2"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__title"
    >
      <h3
        class="pf-c-title pf-m-lg"
        data-ouia-component-id="OUIA-Generated-Title-5"
        data-ouia-component-type="PF4/Title"
        data-ouia-safe="true"
      >
        Insights Ansible Automation Platform
      </h3>
    </div>
    <div
      class="pf-c-card__body pf-u-pb-sm"
    >
      <section
        class=""
        style="background-color: white; min-height: 100%;"
      >
         
        <div
          class="pf-l-bullseye"
        >
          <svg
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-xl"
            role="progressbar"
            viewBox="0 0 100 100"
          >
            <circle
              class="pf-c-spinner__path"
              cx="50"
              cy="50"
              fill="none"
              r="45"
            />
          </svg>
        </div>
         
      </section>
    </div>
  </article>
</DocumentFragment>
`;
